src/lando/main/scm/git.py

```diff
@@ -0,0 +1,298 @@
+        os.environ[ENV_COMMITTER_NAME] = requester_email
+        os.environ[ENV_COMMITTER_EMAIL] = requester_email
+        logger.debug(
+            f"{ENV_COMMITTER_EMAIL} and {ENV_COMMITTER_NAME} set to {requester_email}"
+        )
+        try:
+            yield self
+        finally:
+            del os.environ[ENV_COMMITTER_NAME]
+            del os.environ[ENV_COMMITTER_EMAIL]
+
+    def head_ref(self) -> str:
+        """Get the current revision_id"""
+        return self._git_last_commit_id()
+
+    def changeset_descriptions(self) -> list[str]:
+        """Retrieve the descriptions of commits in the repository.
+
+        Returns:
+            list[str]: A list of first lines of changeset descriptions.
+        """
+        command = ["log", "--format=%s", "@{u}.."]
+        return self._git_run(*command).splitlines()
+
+    def update_repo(self, pull_path: str, target_cset: Optional[str] = None) -> str:
+        """Update the repository to the specified changeset.
+
+        This method uses the Git commands to update the repository
+        located at the given pull path to the specified target changeset.
+
+        Args:
+            pull_path (str): The path to pull from.
+            target_cset (str): The target changeset to update the repository to.
+
+        Returns:
+            str: The target changeset
+        """
+        branch = target_cset or self.default_branch
+        self.clean_repo()
+        self._git_run("pull", "--prune", pull_path)
+        self._git_run("checkout", "--force", "-B", branch)
+
+    def clean_repo(self, *, strip_non_public_commits=True):
+        """Reset the local repository to the origin"""
+        if strip_non_public_commits:
+            self._git_run("reset", "--hard", "origin/HEAD")
+        self._git_run("clean", "-fdx")
+
+    def format_stack_amend(self) -> Optional[list[str]]:
+        """Amend the top commit in the patch stack with changes from formatting."""
+        self._git_run("commit", "--all", "--amend", "--no-edit")
+        return [self.get_current_node()]
+
+    def format_stack_tip(self, commit_message: str) -> Optional[list[str]]:
+        """Add an autoformat commit to the top of the patch stack.
+
+        Return the commit hash of the autoformat commit as a `str`,
+        or return `None` if autoformatting made no changes.
+        """
+        self._git_run("commit", "--all", "--message", commit_message)
+        return [self.get_current_node()]
+
+    def get_current_node(self):
+        """Return the commit_id of the tip of the current branch"""
+        return self._git_run("rev-parse", "HEAD").stdout.strip()
+
+    @property
+    def repo_is_initialized(self) -> bool:
+        """Determine whether the target repository is initialised"""
+        return Path(self.path).exists() and self._git_call("status", cwd=self.path)
+
+    @classmethod
+    def repo_is_supported(self, path: str) -> bool:
```

Comment:
`path` is really more of a `remote_url`.

Reply:
Yes and no: it can actually be a local git repo as well (that's how we test it (:)